pkg/controllers/provisioningv2/cluster/remove.go

```diff
@@ -78,9 +100,12 @@ func (h *handler) doClusterRemove(cluster *v1.Cluster) (string, error) {
 			return "", err
 		}
 
-		_, err = h.mgmtClusterCache.Get(cluster.Status.ClusterName)
-		if !apierrors.IsNotFound(err) {
-			return fmt.Sprintf("waiting for cluster [%s] to delete", cluster.Status.ClusterName), nil
+		if h.isLegacyCluster(cluster) {
+			// If this is a legacy cluster (i.e. RKE1 cluster) then we should wait to remove the provisioning cluster until the v3.Cluster is gone.
```

Comment:
Why don't we want to wait with a non-legacy cluster? There's still a corresponding v3 Cluster, right?

Reply:
Yes, but it depends on which cluster object will get deleted first.

For legacy clusters, provisioning cluster object gets cleaned up faster, so the provisioning cluster will wait for the management cluster.

For the new provisioning, the management cluster will get cleaned up faster, so the management cluster will wait for the provisioning cluster.